src/KeyringController.test.ts

```diff
@@ -37,26 +37,59 @@ const walletTwoAddresses = [
   '0x49dd2653f38f75d40fdbd51e83b9c9724c87f7eb',
 ];
 
-describe('KeyringController', () => {
-  let keyringController: KeyringController;
-
-  beforeEach(async () => {
-    keyringController = new KeyringController({
-      encryptor: mockEncryptor,
-      cacheEncryptionKey: false,
-      keyringBuilders: [keyringBuilderFactory(KeyringMockWithInit)],
-    });
+/**
+ * Create a keyring controller that has been initialized with the given options.
+ *
+ * @param options - Initialization options.
+ * @param options.constructorOptions - Constructor options, merged with test defaults.
+ * @param options.password - The vault password. If provided, creates a new vault (if necessary)
+ * and unlocks the vault.
+ * @param options.seedPhrase - A seed phrase. If provided, this is used to restore the vault.
+ * @returns A keyring controller.
+ */
+async function initializeKeyringController({
+  constructorOptions,
+  password,
+  seedPhrase,
+}: {
+  constructorOptions?: Partial<
+    ConstructorParameters<typeof KeyringController>[0]
+  >;
+  password?: string;
+  seedPhrase?: string;
+} = {}) {
+  const keyringController = new KeyringController({
+    encryptor: mockEncryptor,
+    cacheEncryptionKey: false,
+    keyringBuilders: [keyringBuilderFactory(KeyringMockWithInit)],
+    ...constructorOptions,
+  });
 
+  if (seedPhrase && !password) {
+    throw new Error('Password required to restore vault');
+  } else if (seedPhrase && password) {
+    await keyringController.createNewVaultAndRestore(
+      PASSWORD,
+      walletOneSeedWords,
+    );
+  } else if (password) {
     await keyringController.createNewVaultAndKeychain(PASSWORD);
     await keyringController.submitPassword(PASSWORD);
```

Comment:
Nit: this should not be needed as unlock is already done inside `createNewVaultAndKeychain`

Reply:
Huh, good point. I was just copying this from the pre-existing `beforeEach`. But yes we can remove this. Done in 7195eb1